interpreters/src/table_manipulator/meta_based.rs

```diff
@@ -45,21 +48,23 @@ impl TableManipulator for TableManipulatorImpl {
                 ),
             })?;
 
-        let partition_info = plan.partition_info.map(|v| {
+        let partition_info = plan.partition_info.clone().map(|v| {
             v.get_definitions()
                 .iter()
                 .map(|def| format_sub_partition_table_name(&plan.table, &def.name))
                 .collect::<Vec<String>>()
         });
 
+        let encoder = PartitionInfoEncoder::default();
         let req = CreateTableRequest {
             schema_name: ctx.default_schema().to_string(),
             name: plan.table,
             encoded_schema,
             engine: plan.engine,
             create_if_not_exist: plan.if_not_exists,
             options: plan.options,
-            partition_info: partition_info.map(|v| PartitionInfo { names: v }),
+            partition_table_info: partition_info.map(|v| PartitionTableInfo { sub_table_names: v }),
+            encoded_partition_info: encoder.encode(plan.partition_info.unwrap()).unwrap(),
```

Comment:
Error from `encode` can't be ignore by `unwrap`.
And I think the logic should not be placed here, it make the initialization too complex.